plugins/jobs/src/js/JobsOverview.tsx

```diff
@@ -0,0 +1,165 @@
+import * as React from "react";
+import Router, {
+  InjectedRouter,
+  LocationDescriptor,
+  Link,
+  withRouter
+} from "react-router";
+
+// tslint:disable-next-line:no-submodule-imports
+import { BehaviorSubject } from "rxjs/BehaviorSubject";
+// tslint:disable-next-line:no-submodule-imports
+import "rxjs/add/observable/empty";
+
+import { graphqlObservable, componentFromStream } from "data-service";
+import gql from "graphql-tag";
+
+// import JobsTabLoading from "./components/JobsTabLoading";
+// import JobsTabError from "./components/JobsTabError";
+// import JobsTabList from "./components/JobsTabList";
+// import JobsTabEmpty from "./components/JobsTabEmpty";
+
+import JobModel from "./data/JobModel";
+import { combineLatest } from "rxjs/observable/combineLatest";
+import { Observable, Subscribable } from "rxjs/Observable";
+import { JobConnection } from "#PLUGINS/jobs/src/js/types/JobConnection";
+import { Job } from "#PLUGINS/jobs/src/js/types/Job";
+
+interface JobsOverviewProps {
+  path: string[];
+  router: InjectedRouter;
+  location: LocationDescriptor;
+  params: Router.Params;
+}
+
+// const withRouter = getContext({
+//   router: routerShape
+// });
+
+function DummyComponent(props: any): JSX.Element {
+  function updateFilter(event: React.FormEvent<HTMLInputElement>) {
+    props.handleFilterChange(event.currentTarget.value);
+  }
+
+  console.log("rendering with props", props, props.pathname);
+  return (
+    <div style={{ background: "white" }}>
+      <p>Path</p>
+      <p>{props.data && props.data.path.join("/")}</p>
+      <p>Filter</p>
+      <input value={props.filter} onChange={updateFilter} />
+      <p>Jobs</p>
+      {props.data &&
+        props.data.nodes.map((job: Job) => (
+          <p key={job.id}>
+            <Link
+              to={`/jobs/overview2/${props.data.path
+                .concat([job.name])
+                .join(".")}`}
+            >
+              {`link: ${job.name}`}
+            </Link>
+          </p>
+        ))}
+      <p>Children</p>
+      <div>{props.children}</div>
+    </div>
+  );
+}
+
+const JobsOverview = withRouter(
+  componentFromStream<JobsOverviewProps>(
+    (
+      props$: Subscribable<JobsOverviewProps>
+    ): Subscribable<React.ReactNode> => {
+      const jobsOverviewQuery: any = gql`
+        query {
+          jobs(
+            path: $path
+            filter: $filter
+            sortBy: $sortBy
+            sortDirection: $sortDirection
+          ) {
+            path
+            filteredCount
+            totalCount
+            nodes
+          }
+        }
+      `;
+
+      const filter$ = new BehaviorSubject<string>("");
+      const path$ = new BehaviorSubject<string[]>([]);
+
+      (props$ as Observable<JobsOverviewProps>)
```

Comment:
it should actually work as you push props 🤔 no `componentDidMount` should be needed

Reply:
it does work when you're navigating between namespaces, but it brakes once you hit "overview root" :(

Example:

imagine there is a job with the full id: `foo.bar.baz.jobname` and you click trough the namespaces, when you navigate back in browser history, `baz -> bar` works, `bar -> foo` also works, but `foo -> /` breaks :( I'll investigate this after I'm done going through our PRs.

Reply:
So the router doesn't push props to the `JobsOverview` on the `overview2` route because there's nothing to push. It does push `:path` partial when you hit its child route as it should

Reply:
Because of that `props$` never get anything when you go back to `overview2`

Reply:
hm ok, so I need to change the route in `jobs.js` file to also match "with not path"

Reply:
fixed 👍